app/javascript/components/CaseNoteForm.jsx

```diff
@@ -0,0 +1,357 @@
+    }
+
+    let dialog;
+    if (this.state.type === "create" || this.state.type === "edit") {
+      dialog = (
+        <Dialog
+          style={styles.dialogStyle}
+          open={this.state.open}
+          onClose={this.handleClose}
+          aria-labelledby="form-dialog-title"
+          maxWidth="sm"
+          fullWidth
+        >
+          <DialogContent maxwidth="sm">
+            <DialogContentText style={styles.dialogContentTextStyle}>
+              Title
+            </DialogContentText>
+            <TextField
+              value={this.state.type === "create" ? this.state.title : this.state.tempTitle}
+              style={styles.dialogContentTextFieldStyle}
+              name="title"
+              onChange={this.handleChange(title)}
+              variant="outlined"
+              margin="dense"
+              id="title"
+              label="Case Note title"
+              type="text"
+              fullWidth
+            />
+          </DialogContent>
+          <br />
+
+          <DialogContent maxwidth="sm">
+            <DialogContentText style={styles.dialogContentTextStyle}>
+              Description
+            </DialogContentText>
+            <MuiThemeProvider theme={defaultTheme}>
+              <MUIRichTextEditor
+                name="description"
+                value={this.state.type === "create" ? this.state.description.text : this.state.description}
+                onChange={this.handleDescriptionChange(description)}
+                variant="outlined"
+                label="Case Note description"
+                style={styles.MUIRichTextEditorStyle}
+                controls={["bold", "italic", "underline", "numberList", "bulletList", "link"]}
+              />
+            </MuiThemeProvider>
+          </DialogContent>
+          <br />
+
+          <DialogContent>
+            <DialogContentText style={styles.dialogContentTextStyle}>
+              Visible to Participant
+              <Switch
+                name="internal"
+                defaultChecked={false}
+                value={this.state.internal}
+                onChange={this.handleInternalChange(internal)}
+                color="primary"
+                inputProps={{ 'aria-label': 'primary checkbox' }}
+              />
+            </DialogContentText>
+          </DialogContent>
+
+          <DialogActions style={styles.dialogActionsStyle}>
+            <Button
+              onClick={this.handleClose}
+              variant="outlined"
+              color="secondary"
+            >
+              Cancel
+            </Button>
+            <Button
+              onClick={this.handleSubmit}
+              variant="outlined"
+              color="primary"
+            >
+              {this.state.type === "create" ? 'Submit Case Note' : 'Edit Casenote'}
+            </Button>
+          </DialogActions>
+        </Dialog>
+      );
+    }
+    else if (this.state.type === "delete") {
+      dialog = (
+        <Dialog
```

Comment:
Let's create a `DeleteModal` (or something similar) that handles this delete pop-up for us, so we can reuse it in the future! Will send more info

Reply:
Also, you can render the `DeleteModal` directly in the `CaseNoteCard` so that the form doesn't get too messy!

![image](https://user-images.githubusercontent.com/43366398/69905304-a1876200-137f-11ea-8bf8-8287852581b5.png)